tiledb/sm/storage_manager/context.h

```diff
@@ -111,27 +105,63 @@ class Context {
   /** The class logger. */
   shared_ptr<Logger> logger_;
 
+  /** The storage manager. */
+  StorageManager* storage_manager_;
+
   inline static std::atomic<uint64_t> logger_id_ = 0;
 
   /* ********************************* */
   /*         PRIVATE METHODS           */
   /* ********************************* */
 
   /**
-   * Initializes the thread pools.
+   * Initializes the context with the input config. Called only from
+   * constructor.
+   */
+  Status init(const Config& config);
```

Comment:
If it's called only in the constructor, this body of this function should just appear in the constructor.

Reply:
I did this deliberately.  `init()` encapsulates a couple of sub-calls that can fail (return not OK) and bundling them into `init()` only requires one check for failure and corresponding throws and calls to `cleanup_storage_manager()`.  If the body of `init()` is moved into the constructor there will be at least two (highly redundant) checks and error handling.  Unless we bundle those things into a single lambda -- which is what `init()` is.  In fact, there would be three such checks as the return value of `StorageManger::init()` is not checked (but should be).